nixos/tests/nixos-test-driver/lib-extend.nix

```diff
@@ -0,0 +1,27 @@
+let
+  pkgs = import ../../.. {
+    overlays = [
+      (new: old: {
+        lib = old.lib.extend (self: super: {
+          sorry_dave = "sorry dave";
+        });
+      })
+    ];
+  };
```

Comment:
This relies on `builtins.currentSystem` instead of `stdenv.*Platform`, so it breaks `--argstr system foo-linux -A tests.nixos-test-driver.lib-extend`.
You could perhaps use the `pkgs.extend` function to add the overlay. That seems like a simpler solution than getting the parameters and mangling them back into a `localSystem`/`crossSystem` pair. (Or we could make Nixpkgs take *Platform parameters like NixOS does nowadays...)

Reply:
Oh yeah you're right! I completely forgot about `pkgs.extend`! I was focused on my use-case that made use of overlays, but pkgs.extend just works the same here.